test/RewardsStreamerMP.t.sol

```diff
@@ -1449,3 +1449,192 @@ contract UnstakeTest is StakeTest {
+        // ~44.93e18
+        uint256 expectedAccountMP = initialAccountMP + accruedMP - proratedBonusMP + newBonusMP;
+
+        checkAccount(
+            CheckAccountParams({
+                account: vaults[alice],
+                rewardBalance: 0,
+                stakedBalance: stakeAmount,
+                rewardIndex: 0,
+                accountMP: expectedAccountMP,
+                maxMP: 80e18
+            })
+        );
+
+        // 4. Advance time 1 year
+        vm.warp(block.timestamp + 365 days);
+        streamer.updateGlobalState();
+        streamer.updateAccountMP(vaults[alice]);
+
+        accruedMP = _calculeAccuredMP(stakeAmount, 365 days);
+
+        checkAccount(
+            CheckAccountParams({
+                account: vaults[alice],
+                rewardBalance: 0,
+                stakedBalance: stakeAmount,
+                rewardIndex: 0,
+                accountMP: expectedAccountMP + accruedMP,
+                maxMP: 80e18
+            })
+        );
+    }
+
+    function test_LockAfterExpiredLock() public {
+        // Setup - alice stakes 10 tokens with 1 year lock
+        uint256 stakeAmount = 10e18;
+        uint256 initialLockPeriod = 365 days;
+        _stake(alice, stakeAmount, initialLockPeriod);
+
+        uint256 initialBonusMP = _calculateBonusMP(stakeAmount, initialLockPeriod);
+        uint256 initialAccountMP = stakeAmount + initialBonusMP;
+        uint256 initialMaxMP = stakeAmount * streamer.MAX_MULTIPLIER() + stakeAmount + initialBonusMP;
+
+        // Advance time past lock period
+        vm.warp(block.timestamp + initialLockPeriod + 1 days);
+
+        streamer.updateGlobalState();
+        streamer.updateAccountMP(vaults[alice]);
+
+        // Lock for 2 more years
+        uint256 newLockPeriod = 2 * 365 days;
+        uint256 expectedNewBonusMP = _calculateBonusMP(stakeAmount, newLockPeriod);
+
+        _lock(alice, newLockPeriod);
+
+        // Calculate MP accrued during the initial lock period
+        uint256 expectedAccruedMP = _calculeAccuredMP(stakeAmount, initialLockPeriod + 1 days);
+
+        // Check updated state
+        checkAccount(
+            CheckAccountParams({
+                account: vaults[alice],
+                rewardBalance: 0,
+                stakedBalance: stakeAmount,
+                rewardIndex: 0,
+                accountMP: initialAccountMP + expectedNewBonusMP + expectedAccruedMP,
+                maxMP: initialMaxMP + expectedNewBonusMP
+            })
+        );
+    }
+
+    function test_LockFailsWithNoStake() public {
+        vm.expectRevert(RewardsStreamerMP.StakingManager__InsufficientBalance.selector);
+        _lock(alice, 365 days);
+    }
+
+    function test_LockFailsWithInvalidPeriod() public {
+        _stake(alice, 10e18, 0);
+
+        // Test with period = 0
+        vm.expectRevert(RewardsStreamerMP.StakingManager__InvalidLockingPeriod.selector);
+        _lock(alice, 0);
+    }
+}
```

Comment:
If we keep the revert in `lock()` in case there was already a lock in the past, then I need to update/remove some of these tests.